docs/pages/radiogroup.js

```diff
@@ -0,0 +1,695 @@
+      `}
+      />
+
+      <GeneratedPropTable generatedDocGen={generatedDocGen.RadioGroup} />
+      <GeneratedPropTable
+        Component={RadioGroup?.RadioButton}
+        name="RadioGroup.RadioButton"
+        id="RadioGroup.RadioButton"
+        generatedDocGen={generatedDocGen.RadioGroupButton}
+      />
+
+      <MainSection name="Usage guidelines">
+        <MainSection.Subsection columns={2}>
+          <MainSection.Card
+            cardSize="md"
+            type="do"
+            title="When to use"
+            description={`
+          - In a list, form or a Table, to present users with multiple, related options where only one option can be selected.
+          - When selection doesn’t take immediate effect and requires Form submission.
+        `}
+          />
+          <MainSection.Card
+            cardSize="md"
+            type="don't"
+            title="When not to use"
+            description={`
+          - Situations where users can select multiple options. Use [Checkbox](/checkbox) instead.
+          - When there is only one item to select or deselect. Use Checkbox instead.
+          - When a selection takes immediate effect, especially on mobile. Use [Switch](/switch) instead.
+          - When visually, it’s hard to tell that a RadioGroup turns something on or off. Use Switch instead.
+        `}
+          />
+        </MainSection.Subsection>
+      </MainSection>
+      <MainSection name="Best Practices" />
+      <MainSection name="Accessibility">
+        <MainSection.Subsection
+          title="Labels"
+          description="Each RadioButton in a RadioGroup should have a label that can be read by screen readers, and that can be clicked or tapped to make it easier for users to select and deselect options. Therefore, make sure to supply the label prop. If that’s not possible, make sure your stand-alone Label has an `htmlFor` prop that matches the `id` of the RadioButton. Test that a RadioButton and label are properly connected by clicking or tapping on the label and confirming that it activates the RadioButton next to it."
+        />
+        <MainSection.Subsection
+          title="Legends"
+          description={`Each RadioGroup should have a legend that clearly delineates what is being chosen. If you cannot use the provided legend styling, \`legendDisplay\` can be set to \`hidden\`, and an alternative legend can be displayed.`}
+        />
+        <MainSection.Subsection
+          title="Keyboard interaction"
+          description={`After focus has been set on the first RadioButton inside a RadioGroup, the arrow keys are used to cycle focus between the various options. Clicking or tapping the label of RadioButton should also focus that particular RadioButton. `}
+        />
+        <MainSection.Subsection
+          title="Error message"
+          description={`If RadioGroup has an \`errorMessage\`, the \`id\` on RadioGroup must also be set to properly link the error message to RadioGroup.
```

Comment:
Again, I think `id` should just be required regardless. Completely obviates this type of error.